model/DataObject.php

```diff
@@ -1575,6 +1575,11 @@ public function getComponents($componentName, $filter = "", $sort = "", $join =
 			);
 		}
 
+		if($filter !== null || $sort !== null || $limit !== null) {
```

Comment:
`$join` already throws an `InvalidArgumentException` just above here, so I figured I’d leave that untouched!